src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -72,70 +79,101 @@ public static Level ParseLevel(string levelToParse)
-            }
+            HashSet<string> rv = new HashSet<string>();
 
-            foreach (string dll in dlls)
+            foreach (var item in getValidTestSuiteTypesFromAssembly(assemblyFullPath))
             {
-                List<Type> testSuites = getValidTestSuites(dll);
+                Level lv = ParseLevel(_options.Level);
 
-                foreach (Type testSuite in testSuites)
+                switch (lv)
                 {
-                    getValidTestsFromTestSuite(testSuite, level);
+                    case Level.Namespace:
+                        rv.Add(item.Namespace);
+                        break;
+                    case Level.Class:
+                        rv.Add(item.FullName);
+                        break;
+                    case Level.Function:
+                        rv.IntersectWith(getValidTestsFromTestSuite(item));
+                        break;
+                    case Level.TestCase:
+                        break;
+                    default:
+                        throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, "Invalid level option: {0}", lv.ToString()));
                 }
             }
+
+            return rv;
         }
 
         /// <summary>
-        /// Get all valid test suites from the assembly
+        /// 
         /// </summary>
-        /// <param name="assemblyFullPath">The full qualified path for the assembly</param>
-        /// <returns>Return a list of test suites</returns>
-        private static List<Type> getValidTestSuites(string assemblyFullPath)
+        /// <param name="assemblyFullPath"></param>
+        /// <returns></returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
+        private static Assembly loadAssembly(string assemblyFullPath)
         {
             if (string.IsNullOrWhiteSpace(assemblyFullPath))
             {
-                throw new ArgumentNullException("Assembly full path can't be NULL or Empty.");
+                throw new ArgumentNullException("assemblyFullPath", "can't be NULL or Empty.");
             }
 
             Assembly assembly = null;
+
             try
             {
                 assembly = Assembly.LoadFrom(assemblyFullPath);
             }
             catch (FileNotFoundException)
             {
-                string errorMessage = string.Format("File was not found: '{0}'", assemblyFullPath);
+                string errorMessage = string.Format(CultureInfo.CurrentCulture, "File was not found: '{0}'", assemblyFullPath);
 
                 throw new FileNotFoundException(errorMessage);
             }
 
-            List<Type> testSuites = new List<Type>();
-            foreach (var item in assembly.GetTypes())
+            return assembly;
+        }
+
+        /// <summary>
+        /// 
```

Comment:
Missing documentation.  Same with param & returns below.

Reply:
Will add that